cmd/podman/images/scp.go

```diff
@@ -0,0 +1,317 @@
+			rep, err := execCommand(iden[1], uri[1], scpOpts.Save.Output, scpOpts.Load.Input, scpOpts.ImageName, scpOpts.Tag, true) // execute the command from line 207 with ToFrom defaulting to true
+			if err != nil {
+				return err
+			}
+			fmt.Println(rep)
+			break
+		}
+		report, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(report.Names, ","))
+	case scpOpts.ToRemote: // remote host load
+		scpOpts.Save.Format = "oci-archive"
+		abiErr := abiEng.Save(context.Background(), args[0], tags, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+		rep, err := execCommand(iden[0], uri[0], scpOpts.Save.Output, scpOpts.Load.Input, args[0], "", scpOpts.ToRemote)
+		if err != nil {
+			return err
+		}
+		fmt.Println(rep)
+	// TODO: Add podman remote support
+	default: // else native load
+		scpOpts.Save.Format = "oci-archive"
+		abiErr := abiEng.Save(context.Background(), args[0], tags, scpOpts.Save) // save the image locally before loading it on remote, local, or ssh
+		if abiErr != nil {
+			errors.Wrapf(abiErr, "could not save image as specified")
+		}
+		rep, err := abiEng.Load(context.Background(), scpOpts.Load)
+		if err != nil {
+			return err
+		}
+		fmt.Println("Loaded images(s): " + strings.Join(rep.Names, ","))
+	}
+	err = os.Remove(f.Name())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// execCommand takes ssh information and an image file path on the client, connects via ssh, executes
+// an scp command and then calls podman image load on the remote machine given the new scp filepath
+func execCommand(iden string, uri *urlP.URL, imageFileIn string, imageFileOut string, image string, tag string, LoadSave bool) (string, error) {
+	cfg, err := connection.Validate(uri, iden)
+	if err != nil {
+		return " ", err
+	}
+	dial, err := ssh.Dial("tcp", uri.Host, cfg) // dial the client
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to connect")
+	}
+	defer dial.Close()
+
+	session, err := dial.NewSession() // connect to the dialed client
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create new ssh session on %q", uri.Host)
+	}
+	defer session.Close()
+	if err != nil {
+		return "", err
+	}
+
+	if LoadSave { // if we are loading to the remote/ssh client
+		f, _ := os.Open(imageFileIn) // open image file to copy
+		defer f.Close()
+		n, err := scpD.CopyTo(dial, imageFileIn, imageFileOut)
+		if err != nil {
+			fmt.Println(n, "Bytes copied before error. Error while copying file ", err)
+		}
+
+		podman := os.Args[0]
+		if v, found := os.LookupEnv("PODMAN_BINARY"); found {
+			podman = v // get path to podman bin
+		}
+		run := ""
+		if tag != "" {
+			run = podman + " image load --input=" + imageFileOut + ";" + podman + " tag " + scpOpts.ImageName + " " + tag // run ssh image load of the file copied via scp
```

Comment:
This isn’t a proper implementation of the “load using a different name” feature:
- It overwrites the original-name tag on the destination, which the user didn’t ask for
- (And the original-name tag is left around)
- The `!ToRemote` code path also needs to provide the “load using a different name” feature, doesn’t it?

I’d be perfectly fine with a first version that just does `if sourceImageName != destImageName { fail ("this is currently not implemented")`. 

---

It’s also surprising and a bit hard to track if a function directly references a global like this. Have the caller pass that as a parameter.

Reply:
Good point, I will take out the image naming convention for a v1, I might be able to use podman tag in a clever way so that it doesn't overwrite but simpler is better for now

Reply:
It’s the load that needs to be instructed to use a user-chosen name for the loaded image. After that loads the image using the original name, it’s too late.